creator-node/src/snapbackSM/computeSyncModeForUserAndReplica.js

```diff
@@ -0,0 +1,116 @@
+const DBManager = require('../dbManager.js')
+const { asyncRetry } = require('../utils.js')
+
+/**
+ * Sync mode for a (primary, secondary) pair for a user
+ */
+const SyncMode = Object.freeze({
+  None: 'NONE',
+  SecondaryShouldSync: 'SECONDARY_SHOULD_SYNC',
+  PrimaryShouldSync: 'PRIMARY_SHOULD_SYNC'
+})
+
+const FetchFilesHashNumRetries = 3
+
+/**
+ * Given user state info, determines required sync mode for user and replica. This fn is called for each (primary, secondary) pair
+ * @notice Is used when both replicas are running version >= 0.3.51
+ * @param {Object} param
+ * @param {string} param.wallet user wallet
+ * @param {number} param.primaryClock clock value on user's primary
+ * @param {number} param.secondaryClock clock value on user's secondary
+ * @param {string} param.primaryFilesHash filesHash on user's primary
+ * @param {string} param.secondaryFilesHash filesHash on user's secondary
+ * @returns {SyncMode} syncMode one of None, SecondaryShouldSync, PrimaryShouldSync
+ */
+async function computeSyncModeForUserAndReplica({
+  wallet,
+  primaryClock,
+  secondaryClock,
+  primaryFilesHash,
+  secondaryFilesHash,
+  logger
+}) {
+  if (
+    !(primaryClock && primaryClock !== 0) ||
+    !(secondaryClock && secondaryClock !== 0) ||
```

Comment:
when the pc or sc is == 0, wouldn't this be a valid clock state to enqueue a primary or secondary sync?

Reply:
yes it would, thats why i'm checking here that it is falsey but NOT 0 aka if clock = 0 this condition would be false

Reply:
if we're trying to determine if the clock value is an appropriate integer value, can we actually use `Number.isInteger()` call? [reference](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isInteger#examples)
![image](https://user-images.githubusercontent.com/60366641/151216959-57bd2cb5-f581-49dd-91bd-1f0a911bb0ee.png)

Reply:
also for the negative integer case:
```
Number.isInteger(-100000);   // true
```

Reply:
ahhh dope this is much better, changed